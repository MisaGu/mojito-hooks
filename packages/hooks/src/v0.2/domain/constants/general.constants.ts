export const config = {
  CONTENTFUL_URL: process.env.CONTENTFUL_URL || '',
  CONTENTFUL_AUTH_TOKEN: process.env.CONTENTFUL_AUTH_TOKEN,
  MOJITO_API_URL: process.env.MOJITO_API_URL || '',
  INVOICE_URL: process.env.INVOICE_URL || '',
  ORGANIZATION_ID: process.env.API_ORGANIZATION_ID || '8fb128bd-f55d-4bcc-8b6c-0beb684e4d4e',
  MARKETPLACE_ID: process.env.MARKETPLACE_ID || 'c5ac4f84-b78b-4cb7-a24f-7f0c38da0eb2',
  TEST_ENV: process.env.TEST_ENV || '',
  AUTH0_DOMAIN: process.env.AUTH0_DOMAIN || '',
  AUTH0_CLIENTID: process.env.AUTH0_CLIENTID || '',
  AUTH_REDIRECT_URI: process.env.AUTH_REDIRECT_URI || '',
};
