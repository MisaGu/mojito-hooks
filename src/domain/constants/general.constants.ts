export const config = {
  CONTENTFUL_URL:
    'https://graphql.contentful.com/content/v1/spaces/fu9did2d8yaw/environments/staging' ||
    process.env.CONTENTFUL_URL ||
    '',
  CONTENTFUL_AUTH_TOKEN:
    '19vUSnF3_8S-OsepxXBcDAI_Ua3GbwSy5c7HNTXB-R0' || process.env.CONTENTFUL_AUTH_TOKEN || '',
  MOJITO_API_URL: 'https://api-dev.mojito.xyz/query' || process.env.MOJITO_API_URL || '',
  INVOICE_URL: 'https://sothebys.dev.mojito.xyz/invoice' || process.env.INVOICE_URL || '',
  MARKETPLACE_ID: 'd15b1b02-3c87-434d-a7fb-004b27415fbc' || process.env.MARKETPLACE_ID || '',
  ORGANIZATION_ID: 'a1dc174d-abf4-47d3-be3c-9281c6eba456' || process.env.ORGANIZATION_ID || '',
  AUTH0_DOMAIN: 'https://accounts.staging.sothelabs.com' || process.env.AUTH0_DOMAIN || '',
  AUTH0_CLIENTID: 'thTFyyXuq94Zr6nBIfEVnQ5iUwRlJsSg' || process.env.AUTH0_CLIENTID || '',
  AUTH_REDIRECT_URI: 'http://localhost:3000' || process.env.AUTH_REDIRECT_URI || '',
};
