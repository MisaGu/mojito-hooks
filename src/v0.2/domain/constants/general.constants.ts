export const config = {
  CONTENTFUL_URL: process.env.CONTENTFUL_URL || '',
  CONTENTFUL_AUTH_TOKEN: process.env.CONTENTFUL_AUTH_TOKEN || '',
  MOJITO_API_URL: process.env.MOJITO_API_URL || '',
  INVOICE_URL: process.env.INVOICE_URL || '',
  MARKETPLACE_ID: process.env.MARKETPLACE_ID || '',
  ORGANIZATION_ID: process.env.ORGANIZATION_ID || '',
  AUTH0_DOMAIN: process.env.AUTH0_DOMAIN || '',
  AUTH0_CLIENTID: process.env.AUTH0_CLIENTID || '',
  AUTH_REDIRECT_URI: process.env.AUTH_REDIRECT_URI || '',
};
