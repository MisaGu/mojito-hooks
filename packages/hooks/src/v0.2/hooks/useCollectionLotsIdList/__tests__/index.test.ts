import { act, renderHook } from '@testing-library/react-hooks';
import { DEMO_SLUG } from '../../../components/demo/constants/demo.constants';
import { TestWrapper } from '../../../components/test/wrapper/TestWrapper';
import {
  useCollectionLotsIdList,
  UseCollectionLotsIdListProps,
  UseCollectionLotsIdListReturn,
} from '../useCollectionLotsIdList';

describe.only('useCollectionLotsIdList()', () => {
  it('should be defined', () => {
    expect(useCollectionLotsIdList).toBeDefined();
  });

  it('returns the right result when needed', async () => {
    const { result, rerender, waitFor } = renderHook<
      [UseCollectionLotsIdListProps],
      UseCollectionLotsIdListReturn
    >(
      (args) => {
        return useCollectionLotsIdList(...args);
      },
      {
        initialProps: [{ slug: DEMO_SLUG }],
        wrapper: TestWrapper as any,
      },
    );

    // Trigger request...
    await act(async () => rerender());

    // ...and wait for the main query to start fetching...:
    await waitFor(() => result.current.queryResult.isFetching);

    // ...and for its response:
    await waitFor(() => result.current.queryResult.isSuccess);

    // Check data matches snapshot:
    expect(result.current).toMatchSnapshot({
      queryResult: {
        dataUpdatedAt: expect.any(Number),
      },
    });
  });
});
