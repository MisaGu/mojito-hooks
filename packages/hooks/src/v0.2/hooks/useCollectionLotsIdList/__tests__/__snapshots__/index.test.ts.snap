// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useCollectionLotsIdList() returns the right result when needed 1`] = `
Object {
  "collectionLotsIds": Array [
    Object {
      "contentfulData": Object {
        "lotId": -1,
        "mojitoId": "NA",
        "slug": "NA",
        "subtitle": "NA",
        "title": "NA",
      },
      "id": "mock-item-id",
      "name": "mock-item",
      "saleType": "Auction",
    },
  ],
  "error": null,
  "isLoading": false,
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": true,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
}
`;
